genkit-tools/telemetry-server/src/localFileTraceStore.ts

```diff
@@ -240,11 +240,41 @@ export class Index {
 
   constructor(private indexRoot: string) {
     // TODO: do something about index getting too big. Delete/forget old stuff, etc.
-    this.currentIndexFile = path.resolve(this.indexRoot, '1.idx');
+    this.currentIndexFile = path.resolve(
+      this.indexRoot,
+      this.newIndexFileName()
+    );
+    fs.mkdirSync(this.indexRoot, { recursive: true });
+  }
+
+  clear() {
+    fs.rmSync(this.indexRoot, { recursive: true, force: true });
     fs.mkdirSync(this.indexRoot, { recursive: true });
-    if (!fs.existsSync(this.currentIndexFile)) {
-      fs.writeFileSync(this.currentIndexFile, '');
+    fs.appendFileSync(
+      this.metadataFileName(),
+      JSON.stringify({ version: currentVersion })
+    );
+  }
+
+  metadataFileName() {
+    return path.resolve(this.indexRoot, 'genkit.metadata');
+  }
+
+  getMetadata(): { version: string } | undefined {
+    if (!fs.existsSync(this.metadataFileName())) {
+      return undefined;
     }
+    return JSON.parse(
+      fs.readFileSync(this.metadataFileName(), { encoding: 'utf8' })
+    );
+  }
+
+  private newIndexFileName() {
+    return `idx_${(Date.now() + '').padStart(17, '0')}.json`;
+  }
+
+  listIndexFiles() {
+    return fs.readdirSync(this.indexRoot);
```

Comment:
metadata file will also be in indexRoot dir? So this is not strictly listing index files?

Reply:
Good catch! I had filter, but must have inadvertently removed it during "cleanup". Added back.